server/src/Application/Validation/Validators/FlashcardSet/CreateFlashcardSetRequestValidator.cs

```diff
@@ -0,0 +1,27 @@
+using FluentValidation;
+using StudyZen.Application.Dtos;
+using StudyZen.Application.Services;
+using StudyZen.Application.Validation.Validators;
+
+namespace StudyZen.Application.Validators;
+
+public class CreateFlashcardSetRequestValidator : BaseValidator<CreateFlashcardSetDto>
+{
+    private readonly ILectureService _lectureService;
+    public CreateFlashcardSetRequestValidator(ILectureService lectureService)
+    {
+        _lectureService = lectureService;
+        RuleFor(flashcardSet => flashcardSet.Name)
+        .NotEmpty()
+        .WithMessage("Name must not be empty!")
+        .MaximumLength(50)
+        .WithMessage("Name must not exceed 50 symbols!");
+        RuleFor(flashcardSet => flashcardSet.LectureId)
+        .Must(IsValidLectureId)
+        .WithMessage("Lecture with the given id does not exist!");
+    }
+    protected bool IsValidLectureId(int? lectureId)
+    {
+        return lectureId is null || _lectureService.GetLectureById(lectureId ?? 0) is not null;
```

Comment:
According to REST validation error is 422 (https://stackoverflow.com/questions/8944168/restful-service-how-to-respond-if-validation-failed), instance not found should be 404. So this should probably be in service

Reply:
I wouldn't agree, because here the client sends an invalid lectureId field, so the status code, in my opinion, should be 422